### PR TITLE
DM-39022: Add group name presence server.

### PR DESCRIPTION
**Pull request by ktlim**
This change records the group names/ids of images as they're being ingested and adds a Web service that reports the presence or absence of a particular group name (with instrument, snap, and detector).

### CODE REVIEW

File: src/ingest.py [hunk 1]
```diff
@@ -162,12 +201,17 @@ def main():
...
+                if not is_lfa:
+                    record_groups(resources)
+
                 logger.info("Ingesting %s", resources)
                 refs = None
                 try:
                     refs = ingester.run(resources)
                 except Exception:
                     logger.exception("Error while ingesting %s", resources)
+                    continue
```
Comment (hsinfang) [inline src/ingest.py:216]: Why is this needed with this commit?

**Review by hsinfang — APPROVED** [review]
LGTM!

File: src/ingest.py (hunk 1 above)
Comment (ktlim) [reply src/ingest.py:216]: This is actually a separate fix that prevents visit definition (and Rucio registration) from trying to run if an exception occurred in ingest.  I'll split it out into a separate commit.
